### PR TITLE
Add version: ShiningLight.OpenSSL version 3.0.10

### PR DESCRIPTION
**Pull request by szhu25**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/114618&drop=dogfoodAlpha

### CODE REVIEW

File: manifests/s/ShiningLight/OpenSSL/3.0.10/ShiningLight.OpenSSL.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  InstallerUrl: https://slproweb.com/download/Win32OpenSSL-3_0_10.msi
+  InstallerSha256: CAC80723B8DA0F5286FF88F9F634EDA76B9B4537E495DD76F6FD3D14E942635F
+  ProductCode: '{391EBF3B-21B5-46A2-A35A-9ACBD11013F5}'
+- Architecture: x64
+  InstallerType: inno
+  InstallerUrl: https://slproweb.com/download/Win64OpenSSL-3_0_10.exe
+  InstallerSha256: 3717465825172D0062F31BA641DED42BA3178F05A762A31199B0ED5F2943B96C
+  ProductCode: OpenSSL (64-bit)_is1
+  AppsAndFeaturesEntries:
+  - DisplayName: OpenSSL 3.0.8 (64-bit)
```
Comment (ItzLevvie) [inline manifests/s/ShiningLight/OpenSSL/3.0.10/ShiningLight.OpenSSL.installer.yaml:36]:
```suggestion
  - DisplayName: OpenSSL 3.0.10 (64-bit)
```

File: manifests/s/ShiningLight/OpenSSL/3.0.10/ShiningLight.OpenSSL.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Publisher: OpenSSL Win64 Installer Team
+    ProductCode: OpenSSL (64-bit)_is1
+- Architecture: x86
+  InstallerType: inno
+  InstallerUrl: https://slproweb.com/download/Win32OpenSSL-3_0_10.exe
+  InstallerSha256: E6C513CEE5A43823D5BDB2218694096A9AA5734E963350B9F00E4CBD132C9C66
+  ProductCode: OpenSSL (32-bit)_is1
+  AppsAndFeaturesEntries:
+  - DisplayName: OpenSSL 3.0.8 (32-bit)
```
Comment (ItzLevvie) [inline manifests/s/ShiningLight/OpenSSL/3.0.10/ShiningLight.OpenSSL.installer.yaml:45]:
```suggestion
  - DisplayName: OpenSSL 3.0.10 (32-bit)
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Comment by stephengillie** [comment]
Hi @szhu25,

2 of the installers (Inno I believe) require a dependency: 
image

(The `x86` installer requires the `32-bit` dependency.)

**Pull request merged**